AsyncDisplayKit/TextKit/ASTextKitRenderer.mm

```diff
@@ -147,7 +148,18 @@ - (void)_calculateSize
   if (isinf(_constrainedSize.width) == NO && [_attributes.pointSizeScaleFactors count] > 0) {
     _currentScaleFactor = [[self fontSizeAdjuster] scaleFactor];
   }
-  
+
+  // If we do not scale, do exclusion, or do custom truncation, we should just use TextKit for a fast-path.
+  BOOL doesNotScale = _currentScaleFactor == 1;
+  // NOTE: This code does not correctly handle if they set `…` with different attributes.
+  BOOL defaultTruncation = _attributes.avoidTailTruncationSet == nil && [_attributes.truncationAttributedString.string isEqualToString:@"\u2026"];
+  BOOL doesNoExclusion = _attributes.exclusionPaths.count == 0;
+  if (doesNotScale && defaultTruncation && doesNoExclusion) {
+    CGSize baseSize = [_attributes.attributedString boundingRectWithSize:_constrainedSize options:NSStringDrawingUsesLineFragmentOrigin | NSStringDrawingTruncatesLastVisibleLine context:nil].size;
+    _calculatedSize = [self.shadower outsetSizeWithInsetSize:baseSize];
+    return;
+  }
+
   __block NSTextStorage *scaledTextStorage = nil;
   BOOL isScaled = [self isScaled];
```

Comment:
@Adlai-Holler Is this variable different or can be shared with `doesNotScale`?

Also, I think _currentScaleFactor always *starts* at 1 — but if pointScaleFactors has other values, and the content doesn't fit the constraint, then other scales are tried.  Is it correct that if _currentScaleFactor is 1 on this pass, that we are not using scaling at all?

Best option would be to find one of our uses of pointScaleFactors in Pinterest and test this with a hardcoded string that triggers the scaling.